lib/puppet/settings.rb

```diff
@@ -891,9 +891,12 @@ def service_group_available?
   # Allow later inspection to determine if the setting was set on the
   # command line, or through some other code path.  Used for the
   # `dns_alt_names` option during cert generate. --daniel 2011-10-18
+  #
+  # @param param [String, Symbol] the setting to look up
+  # @return [Object, nil] the value of the setting or nil if unset
   def set_by_cli?(param)
     param = param.to_sym
-    !@value_sets[:cli].lookup(param).nil?
+    @value_sets[:cli].lookup(param)
```

Comment:
Changing what this method returns could break someone. Maybe define a new method `set_by_cli` that returns the value, and then have `set_by_cli?` and have it call the first one?

Reply:
Updated. I had to also change `set_in_section?` to behave the same (return a boolean), but I think we're safe with that one since it's a fairly new method (I added it in c641b04b0, but its usage was removed in 79737d7f).